ext/mbstring/mb_gpc.c

```diff
@@ -255,6 +243,7 @@ const mbfl_encoding *_php_mb_encoding_handler_ex(const php_mb_encoding_handler_i
 		if (identd != NULL) {
 			n = 0;
 			while (n < num) {
+				mbfl_string string;
```

Comment:
Is this whole `else` part covered by a test?

Reply:
Hi @kamil-tekiela. This function is used to implement a couple of higher-level functions like `mb_parse_str`. Those higher-level functions have various tests, and this section where it tries to detect the encoding of the input string is definitely exercised by tests.

Reply:
Sorry for asking dumb questions, but which test? Or can you give me a MCVE that would allow me to set a breakpoint?

Reply:
@kamil-tekiela I'm starting to doubt whether that block is actually covered by tests. I will check a bit later and confirm.

Reply:
It works with `ini_set('mbstring.http_input', 'UTF-8,SJIS,EUC-JP')` but I get a deprecation warning.

Reply:
Yes, that is expected.

Reply:
@kamil-tekiela!!! You were right!!!

There was no test exercising the case where `_php_mb_encoding_handler_ex` was called with multiple candidate encodings!

I have added one; please see the latest commit added to this PR.

Reply:
Thanks. This looks good now. I think that part of the code was working correctly anyway, but it's good to have a test for it.